cmd/tempo-vulture/main.go

```diff
@@ -186,17 +197,83 @@ func main() {
 			oldest = startTime
 		}
 		newStart, ts = selectPastTimestamp(oldest, now, interval, vultureConfig.tempoRetentionDuration, r)
-		return
+		return newStart, ts, skip
 	}
 
 	selectOldTimestamp := func(now time.Time) (newStart, ts time.Time, skip bool) {
 		newest := now.Add(-vultureConfig.tempoRecentTracesCutoffDuration)
 		if newest.Before(startTime) { // if vulture's just started and no traces to query
 			skip = true
-			return
+			return newStart, ts, skip
 		}
 		newStart, ts = selectPastTimestamp(startTime, newest, interval, vultureConfig.tempoRetentionDuration, r)
-		return
+		return newStart, ts, skip
+	}
+
+	// Check if we're running in validation mode
+	if validationMode {
+		accessPolicyToken := os.Getenv("TEMPO_ACCESS_POLICY_TOKEN")
+		if accessPolicyToken == "" {
+			logger.Error("TEMPO_ACCESS_POLICY_TOKEN environment variable is required in validation mode")
+			os.Exit(1)
+		}
+
+		// Construct the basic auth token for HTTP headers
+		basicAuthToken := constructAuthToken(vultureConfig.tempoOrgID, accessPolicyToken)
+
+		httpClient.SetHeader("Authorization", fmt.Sprintf("Basic %s", basicAuthToken))
+
+		// Create authenticated jaeger client for writing traces
+		authJaegerClient, err := utilpkg.NewJaegerToOTLPExporterWithAuth(
+			pushEndpoint,
+			vultureConfig.tempoOrgID,
+			basicAuthToken,
+			vultureConfig.tempoPushTLS,
+		)
+		if err != nil {
+			logger.Error("Failed to create authenticated OTLP exporter", zap.Error(err))
+			os.Exit(1)
+		}
+
+		logger.Info("Running in validation mode",
+			zap.Int("cycles", validationCycles),
+			zap.Duration("timeout", validationTimeout),
+			zap.String("org_id", vultureConfig.tempoOrgID),
+		)
+
+		ctx := context.Background()
+
+		validationConfig := ValidationConfig{
+			Cycles:                validationCycles,
+			Timeout:               validationTimeout,
```

Comment:
Timeout looks unused.  Consider using a context with timeout so that when you call RunValidation with the context, the cancellation handles the desired timeout.